types/stylis/index.d.ts

```diff
@@ -6,20 +6,20 @@
 export as namespace stylis;
 
 export interface Element {
-    parent?: Element | undefined;
-    children?: Element[] | string | undefined;
-    root: Element;
+    parent: Element | null;
+    children: Element[] | string;
+    root: Element | null;
     type: string;
     props: string[] | string;
     value: string;
     length: number;
     return: string;
-    line?: number | undefined;
-    column?: number | undefined;
+    line: number;
+    column: number;
 }
 
 export type ArrayMapCallback = (value: string, index: number, array: string[]) => string;
-export type Middleware = (element: Element, index: number, children: Array<Element | string>, callback: Middleware) => string | void;
+export type Middleware = (element: Element, index: number, children: Array<Element>, callback: Middleware) => string | void;
```

Comment:
This I'm slightly unsure about - it definitely wasn't correct because if anything this could be `Element[] | string[]` **but** the `string[]` isn't ever passed to a `Middleware` in the whole test suite of Stylis. If one comes up with compelling use case when this receives `string[]` then we can always include that in the Stylis test suite and relax the type definition here.